site/app/views/submission/HomeworkView.php

```diff
@@ -1185,14 +1187,38 @@ public function showRegradeDiscussion(GradedGradeable $graded_gradeable, bool $c
                     $gc_title = $component->getTitle();
                 }
 
+                $instructor_full_access = [];
+                $limited_access_grader = [];
+
+                $grade_inquiry_posts_for_id = $grade_inquiry_posts[$grade_inquiry->getId()];
+                $author_user_ids = array_map(function ($post) {
+                    return $post["user_id"];
+                }, $grade_inquiry_posts_for_id);
+                $author_user_groups = $this->core->getQueries()->getAuthorUserGroups($author_user_ids);
+
+                foreach ($author_user_groups as $author) {
+                    $limited_access_grader[$author["user_id"]] = $author["user_group"] == User::GROUP_LIMITED_ACCESS_GRADER;
+                    $instructor_full_access[$author["user_id"]] = $author["user_group"] <= User::GROUP_FULL_ACCESS_GRADER;
+                }
+
                 // format posts
                 $posts = [];
                 foreach ($grade_inquiry_posts[$grade_inquiry->getId()] as $post) {
                     if (empty($post)) {
                         break;
                     }
                     $is_staff = $this->core->getQueries()->isStaffPost($post['user_id']);
-                    $name = $this->core->getQueries()->getUserById($post['user_id'])->getDisplayedFirstName();
+                    $is_limited_access_grader = $limited_access_grader[$post['user_id']];
+                    $is_instructor_or_full_access_grader = $instructor_full_access[$post['user_id']];
+                    $first_name = $this->core->getQueries()->getUserById($post['user_id'])->getDisplayedFirstName();
```

Comment:
Now I don't know a lot about queries, but if you look in the file Submitty\site\tests\app\libraries\AccessTester.php on line 38, you can perhaps store the information from $this->core->getQueries() in a variable $queries, and then you can use that variable to find information for the variables $is_staff, $first_name, and $last_name.

Reply:
Yeah I agree with William maybe having so many queries in a for loop is not such a good idea. Also it seems this branch has some conflicts that you'll have to resolve. But great job!

Reply:
Also I am getting an error when loading the site

![image](https://user-images.githubusercontent.com/117249183/225124359-532e1f92-adfc-4613-a790-0eb0c1e11ba5.png)